python/pyspark/sql/types.py

```diff
@@ -1674,6 +1674,16 @@ def from_arrow_type(at):
         if types.is_timestamp(at.value_type):
             raise TypeError("Unsupported type in conversion from Arrow: " + str(at))
         spark_type = ArrayType(from_arrow_type(at.value_type))
+    elif types.is_struct(at):
+        # TODO: remove version check once minimum pyarrow version is 0.11.0
+        if LooseVersion(pa.__version__) < LooseVersion("0.11.0"):
+            raise TypeError("Unsupported type in conversion from Arrow: " + str(at) +
+                            "\nPlease install pyarrow >= 0.11.0 for StructType support.")
```

Comment:
Currently only support for pyarrow >=0.11 since I couldn't find a way to reconstruct pandas DataFrame from `pyarrow.lib.StructArray`.